main.go

```diff
@@ -883,6 +884,13 @@ func (task *TaskRun) run() error {
 	if err != nil {
 		return WorkerShutdown(err)
 	}
+	// using NewWriterLevel(...) instead of NewWriter(...) to be explicit about
+	// compression level, so easy to change later
+	gzipLogWriter, err := gzip.NewWriterLevel(logFileHandle, gzip.DefaultCompression)
```

Comment:
So it seems that NewWriterLevel is almost used instead as a documentation piece to let the developer know that the default level can be overridden? I guess if we were actually overriding it via config or something then it makes sense to use this method, but otherwise this is why the simpler (readability wise) method could be used.  Not big deal though.